src/arithmetic/algebraic_expression.c

```diff
@@ -371,11 +371,14 @@ static AlgebraicExpressionOperand _AlgebraicExpression_OperandFromEdge (
         GrB_Matrix_new(&m, GrB_BOOL, Graph_RequiredMatrixDim(g), Graph_RequiredMatrixDim(g));
 
         for(uint i = 0; i < reltype_count; i++) {
-            // TODO does this make sense if the reltype is created by this query?
-            // char *label = astEdge->labels[i];
-            // Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_EDGE);
-            // if(!s) continue;
-            GrB_Matrix l = Graph_GetRelationMatrix(g, e->reltypeIDs[i]);
+            GrB_Matrix l;
+            uint reltype_id = e->reltypeIDs[i];
+            if (reltype_id == GRAPH_UNKNOWN_RELATION) {
+                // No matrix to add
+                continue;
+            } else {
```

Comment:
No need for this `else` in case reltype_id == GRAPH_UNKNOWN_RELATION we'll jump to the beginning of the loop